js/SteamExample.js

```diff
@@ -0,0 +1,189 @@
+
+    var friendAvatarIndex = steamFriends.getSmallFriendAvatar(friendSteamId);
+    var friendAvatarWidth = steamUtils.getImageWidth(friendAvatarIndex);
+    var friendAvatarHeight = steamUtils.getImageHeight(friendAvatarIndex);
+
+    console.log(`SteamUtils - Image ${friendAvatarIndex} - ${friendAvatarWidth}x${friendAvatarHeight}`);
+
+    if (canvas) {
+      var friendAvatarRGBA = steamUtils.getImageRGBA(friendAvatarIndex);
+      var friendAvatarImageArray = convertBase64ToBinary(friendAvatarRGBA);
+      var friendAvatarImageData = new ImageData(friendAvatarImageArray, friendAvatarWidth, friendAvatarHeight);
+
+      ctx.putImageData(friendAvatarImageData, 0, y);
+      ctx.fillStyle = "white";
+      ctx.fillText(`${friendPersonaName} (${friendPersonaState})`, friendAvatarWidth + 10, y + ((friendAvatarHeight / 2) + 10));
+
+      y += friendAvatarHeight;
+    }
+  }
+
+  var ob1 = notificationCenter.addInstanceObserver("SteamUserStats", "NumberOfCurrentPlayersResponse", steamUserStats, function (sender, info) {
+    console.log(`SteamUserStats - NumberOfCurrentPlayersResponse - ${JSON.stringify(info)}`);
+    ob1.release();
+  });
+
+  var achievementCount = steamUserStats.getNumberOfAchievements();
+  var ob2 = notificationCenter.addInstanceObserver("SteamUserStats", "UserStatsReceivedResponse", steamUserStats, function (sender, info) {
+
+    console.log(`SteamUserStats - UserStatsReceivedResponse - ${JSON.stringify(info)}`);
+    steamUserStats.requestGlobalAchievementPercentages();
+
+    var ob3 = notificationCenter.addInstanceObserver("SteamUserStats", "GlobalAchievementPercentagesResponse", steamUserStats, function (sender, info) {
+      console.log(`SteamUserStats - NumberOfAchievements - ${achievementCount}`);
+
+      for (var i = 0; i < achievementCount; i += 1) {
+        var achievementName = steamUserStats.getAchievementName(i);
+        var achievementFriendlyName = steamUserStats.getAchievementDisplayAttribute(achievementName, "name");
+        var achievementDescription = steamUserStats.getAchievementDisplayAttribute(achievementName, "desc");
+        var achievementHidden = steamUserStats.getAchievementDisplayAttribute(achievementName, "hidden");
+        var achievementAchieved = steamUserStats.getAchievement(achievementName);
+        var achievementAchievedPercent = steamUserStats.getAchievementAchievedPercent(achievementName);
+
+        console.log(`SteamUserStats - Achievement ${(i + 1)} - ${achievementName} - "${achievementFriendlyName}" - ${achievementDescription} - ${(achievementHidden === "1") ? "Invisible " : ""}${(achievementAchieved) ? "Achieved ": ""}${achievementAchievedPercent}%`);
+      }
+
+      ob3.release();
+    });
+
+    ob2.release();
+  });
+
+  // Wait for results in notification callbacks
+  steamUserStats.getNumberOfCurrentPlayers();
+  steamUserStats.requestUserStats(mySteamId);
+
+  steamFriends.activateGameOverlayToWebPage("http://google.com/");
+}
```

Comment:
This is an absurdly long function, might want to think about breaking it up. Also, it's a bit light on the comments for you.

Reply:
The name is just copying from Steamworks API. I don't want to change it from what Steamworks provides.

Reply:
`interopLoaded` - I don't think that's from SteamWorks.

Reply:
It's 114 lines long, it's got a lot of nugety goodness.

Reply:
I think the PR makes it look like I created all that code from scratch, but I did not, I only renamed the file.